tests/playwright/src/utility/fixtures.ts

```diff
@@ -30,21 +30,29 @@ export type TestFixtures = {
   navigationBar: NavigationBar;
   welcomePage: WelcomePage;
   page: Page;
+  context: BrowserContext;
   statusBar: StatusBar;
 };
 
 export type FixtureOptions = {
   runnerOptions: RunnerOptions;
 };
 
+let newContext = false;
+
 export const test = base.extend<TestFixtures & FixtureOptions>({
   runnerOptions: [new RunnerOptions(), { option: true }],
   runner: async ({ runnerOptions }, use) => {
     const runner = await Runner.getInstance({ runnerOptions });
     await use(runner);
   },
-  page: async ({ runner }, use) => {
-    await use(runner.getPage());
+  context: async ({ runner }, use) => {
+    newContext = true;
+    await use(runner.getNewContext());
+  },
+  page: async ({ runner, context }, use) => {
+    const page = newContext ? context.pages()[0] : runner.getPage();
+    await use(page);
```

Comment:
in this situation `newContext` can't ever be `false` it will always be `true` because as part of page fixtures it will call `context` fixture since it's a dependency of creating `page` fixture.